staking/scripts/build_verifiable_staking_program.sh

```diff
@@ -5,14 +5,37 @@ set -euo pipefail
 # Root of the repository
 REPO_ROOT=$(git rev-parse --show-toplevel)
 
-echo "Building the image for the staking program"
-docker build --platform linux/x86_64 -t staking-build -f "$REPO_ROOT"/staking/Dockerfile "$REPO_ROOT"/staking
+# Default value for TEST
+TEST=false
+
+# Parse command-line arguments
+# -t: build the staking program for tests
+while getopts "t" opt; do
+    case ${opt} in
+    t)
+        TEST=true
+        ;;
+    \?)
+        echo "Invalid option: -$OPTARG" 1>&2
+        exit 1
+        ;;
+    esac
+done
+
+
+if [ "$TEST" = "true" ]; then
+    echo "Building the image for the staking program test"
+    docker build --platform linux/amd64 --build-arg TEST=true -t staking-build -f "$REPO_ROOT"/staking/Dockerfile "$REPO_ROOT"/staking
+else
+    echo "Building the image for the staking program production"
+    docker build --platform linux/amd64 -t staking-build -f "$REPO_ROOT"/staking/Dockerfile "$REPO_ROOT"/staking
+fi
 
 echo "Building the staking program"
-docker run --platform linux/x86_64 --rm -v "$REPO_ROOT"/staking/artifacts:/artifacts staking-build
+docker run --platform linux/amd64 --rm -v "$REPO_ROOT"/staking/artifacts:/artifacts staking-build
```

Comment:
Why this change?

Reply:
I believe `x86_64` is just an alias for `amd64`. I see that everywhere is using `amd64` now so I thought we go with that.